src/test/euler/staking/StakeableEulerERC4626.t.sol

```diff
@@ -69,6 +72,34 @@ contract StakeableEulerERC4626Test is Test {
         vault.updateStakingAddress(1, recipient);
 
         assertEq(address(vault.stakingRewards()), address(stakingRewards));
+        assertEq(eToken.allowance(address(vault), address(stakingRewards)), type(uint256).max);
+    }
+
+    function testFailNotOwnerStaking() public {
+        vault.stake(1000);
+    }
+
+    function testStaking() public {
+        _setStakingContract();
+        uint256 deposited = 10000;
+        uint256 staked = 1000;
+
+        vm.prank(alice);
+        underlying.approve(address(vault), deposited);
+        vm.prank(alice);
+        vault.deposit(deposited, alice);
```

Comment:
Do you need to prank alice two times? Is Foundry that shitty? 🤔

Reply:
`prank` sets the `msg.sender` for the next call